.github/workflows/justtest.yaml

```diff
@@ -0,0 +1,699 @@
+        run: |
+          git submodule update --init --recursive --depth 1 --jobs 8
+
+      # This little adjustment allows to perform a shallow clone
+      # of the LuaJIT submodule.
+      - name: Strip $(git describe) call from LuaJIT build script
+        if: steps.restore-submodules-from-cache.outputs.cache-hit != 'true'
+        run: |
+          rm third_party/luajit/.git
+          sed -i -e 's/CMAKE_SOURCE_DIR/PROJECT_SOURCE_DIR/' third_party/luajit/cmake/SetVersion.cmake
+
+      # }}} Clone repository (with caching)
+
+      # {{{ Configure and build
+
+      # -DTARANTOOL_VERSION=<...>
+      #
+      # Hardcode a fake tarantool version (but with the real git
+      # commit hash).
+      #
+      # The $(git describe --long HEAD) information is not
+      # available, because a shallow clone is performed above. It
+      # is to speed up the job.
+      #
+      # We anyway don't deploy this build to anywhere, it is
+      # needed only to run tests.
+      #
+      # -DCMAKE_BUILD_TYPE=<...>
+      # ------------------------
+      #
+      # Depends on the requested build type: Debug or
+      # RelWithDebInfo.
+      #
+      # -DENABLE_WERROR=ON
+      # ------------------
+      #
+      # Fail the build on a compiler warning regarding the
+      # tarantool source code (adds -Werror compiler flag).
+      # Not sure about compiler flags for submodules.
+      #
+      # -DTEST_BUILD=ON
+      # ---------------
+      #
+      # Shouldn't be used in any build deployed to the outside.
+      # Simplifies detecting various incorrect situations in
+      # testing. Enables a few additional tests. Doesn't depend
+      # from the build type option.
+      #
+      # * Set default timeout of iproto graceful shutdown to
+      #   infinity (see box.cc for details).
+      # * Abort on a memory leak detected on the fiber GC region.
+      # * Set 0.01 seconds sleep in the synchronous queue worker
+      #   fiber instead of 1 second.
+      # * Set `require('tarantool').build.test_build` to `true`.
+      # * Add ER_TEST_* error codes to box error and corresponding
+      #   test cases.
+      - name: Configure tarantool build
+        run: |
+          COMMIT_ID=$(git rev-parse --short HEAD)
+          cmake .                                                   \
+            -DTARANTOOL_VERSION=3.9.9-entrypoint-9-g${COMMIT_ID}    \
+            -DCMAKE_BUILD_TYPE=${{ matrix.cmake-build-type }}       \
+            -DENABLE_WERROR=ON                                      \
+            -DTEST_BUILD=ON
+
+      # The *-deps targets are needed to build libraries for
+      # luajit-test jobs.
+      - name: Build tarantool and test files
+        run: |
+          make -j $(nproc)             \
+            all                        \
+            lua-Harness-tests-deps     \
+            LuaJIT-tests-deps          \
+            PUC-Rio-Lua-5.1-tests-deps \
+            tarantool-c-tests-deps     \
+            tarantool-tests-deps
```

Comment:
I would combine all dep targets to a single one, like `build-test-deps`.

Reply:
Good idea, thanks! It should be easy to do, I'll try.